hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/DefaultClientConnectionStrategy.java

```diff
@@ -34,15 +35,23 @@
     private volatile boolean disconnectedFromCluster;
     private boolean clientStartAsync;
     private ClientConnectionStrategyConfig.ReconnectMode reconnectMode;
+    private ClusterConnector connector;
+
+    @Override
+    public void init(ClientContext clientContext) {
+        super.init(clientContext);
+        ClientConnectionManagerImpl connectionManager = (ClientConnectionManagerImpl) clientContext.getConnectionManager();
```

Comment:
Assume that I am writing a custom strategy, and also assume that accessing the implementation class is illegal and you only need to use the spi interface, how can I tell the connection manager to connect to the cluster? Will you make the ClusterConnector an spi interface?

Reply:
ConnectionManager is not an API.

Reply:
Currently this ClientConnectionStrategy isn't ready to be used as API because it relies on implementation details.